### PR TITLE
Add file creation method that takes an ACL

### PR DESCRIPTION
**Pull request by carlossanlop**
Fixes https://github.com/dotnet/corefx/issues/41614

As soon as [the PR for directory creation ](https://github.com/dotnet/corefx/pull/41834) is merged, I'll rebase this branch to reduce the number of edited files in this PR.

@JeremyKuhne

### CODE REVIEW

**Comment by carlossanlop** [comment]
Apologies. I meant to submit this as draft. I'll have to add the ugly red labels.

File: src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs [hunk 1]
```diff
@@ -10,6 +10,29 @@ namespace System.IO
...
+        /// <exception cref="ArgumentNullException"><paramref name="fileInfo" /> or <paramref name="fileSecurity" /> is <see langword="null" />.</exception>
+        public static FileStream Create(this FileInfo fileInfo, FileMode mode, FileSystemRights rights, FileShare share, int bufferSize, FileOptions options, FileSecurity fileSecurity)
+        {
+            if (fileInfo == null)
+                throw new ArgumentNullException(nameof(fileInfo));
+
+            if (fileSecurity == null)
+                throw new ArgumentNullException(nameof(fileSecurity));
+
+            return null;
```
Comment (stephentoub) [inline src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs]: We should avoid submitting PRs when they aren't functional, aren't meant for review, etc.

Comment (carlossanlop) [reply src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs]: Yes, apologies. I meant to submit a draft but forgot to select the button when submitting.

Comment (stephentoub) [reply src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs]: No need to apologize. But why submit any pr, even a draft? If no one is supposed to review it, and it doesn't contain anything meaningful to review, what's the benefit of opening a pull request?

Comment (carlossanlop) [reply src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs]: I noticed it has been easier for me to track my commits with a draft PR and have Jeremy help review them. But I have to agree, this empty PR just adds noise. I'll close it and reopen it when I have something meaningful to have reviewed.

Comment (stephentoub) [reply src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs]: Thanks

**Pull request closed without merging**